### PR TITLE
Add cloud v2 creation docs

### PR DESCRIPTION
**Pull request by michaellzc**
This is an illustration of the intended workflow, it's still WIP.

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4379** <a href="https://app.graphite.dev/github/pr/sourcegraph/handbook/4379" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/handbook/4379?utm_source=stack-comment).

File: content/departments/cloud/technical-docs/v2.0/creation_process.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Creating a new [managed instance](./index.md) involves following the steps below.
+For basic operations like accessing an instance for these steps, see [managed instances operations](../operations.md) what if there is some text here.
+
+## Prereq
+
+Follow https://github.com/sourcegraph/deploy-sourcegraph-cloud-controller#installation to install `mgv2`
+
+```sh
+git clone https://github.com/sourcegraph/deploy-sourcegraph-cloud-next
+cd deploy-sourcegraph-cloud-next
```
Comment (michaellzc) [inline content/departments/cloud/technical-docs/v2.0/creation_process.md]:
```suggestion
git clone https://github.com/sourcegraph/deploy-sourcegraph-cloud-dev
cd deploy-sourcegraph-cloud-dev
```

File: content/departments/cloud/technical-docs/v2.0/creation_process.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+### Init deployment artifacts
+
+`mgv2 generate` will
+
+- generate the terraform module and apply the terraform module
+- generate the kustomization manifests and helm override based on output from the terraform module
+
+```sh
+mgv2 generate --project-id $PROJECT_ID --domain $DOMAIN
```
Comment (rafax) [inline content/departments/cloud/technical-docs/v2.0/creation_process.md:59]: Is there a reason why you don't want this in the regular `mg` ?

Comment (michaellzc) [reply content/departments/cloud/technical-docs/v2.0/creation_process.md:59]: it's easier to iterate.

`mgv2` and `mg` is compatible and we can literally copy and paste mgv2 code into `mg` along with some find and replace to update the package name.

**Pull request merged**